tests/tracing/test_dnatracing.py

```diff
@@ -8,105 +8,85 @@
 
 # pylint: disable=protected-access
 
+# keeping target dictionary as may want to use these grains for more tests
 
-def test_get_grain_array(test_dnatracing: pd.DataFrame) -> None:
-    """Test extraction of grain array."""
-    target = {
-        1: np.array(
-            [
-                [0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
-                [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0],
-                [0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0],
-                [0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-            ]
-        ),
-        2: np.array(
-            [
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
-            ]
-        ),
-        3: np.array(
-            [
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0],
-                [0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0],
-                [0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-            ]
-        ),
-        4: np.array(
-            [
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                [0, 0, 1, 1, 1, 1, 1, 0, 0, 0, 0],
-            ]
-        ),
+DILTATE_TARGETS = {
+    1: np.asarray(
```

Comment:
As this is a dictionary could we use more informative keys to describe what the arrays are? Otherwise could just use a list.

Reply:
This just mimics how the arrays are fed from the dilate function. It also makes it easier to compare what we expect from grain 1, 2, 3 etc from the GRAINS array in the conftest. i.e. dnatracing_dilate.grains is a dict of grain_number, grain_array, which we can directly compare to the grain number and grain array from the DILATE_TRAGETS